### PR TITLE
ci: initial draft of gitsplit replacement

### LINKED ISSUE #55: [BUG] Gitsplit
### Observed behavior

Gitsplit does not push tags to target repository

### Expected Behavior

Relevant tags in this repository will exist in the downstream repository after splitting

### Steps to reproduce

Merge a release, and check the downstream repository for the created release tag. It will not exist.

### PR DESCRIPTION
**Pull request by tcarrio**
## This PR

- utilizes git-filter-repo to accomplish the task of gitsplit with support for tagging (thus releases for PHP)

### Related Issues

Fixes #55

### Notes

Still in validation, may require merging to actually confirm though

### CODE REVIEW

File: .github/workflows/split_monorepo.yaml [hunk 1]
```diff
@@ -1,23 +1,131 @@
...
-      - main
-      - split
   release:
     types: [published]
-  create:
-  workflow_dispatch:
 
 jobs:
-  gitsplit:
+  split-hook-dd-trace:
```
Comment (beeme1mr) [inline .github/workflows/split_monorepo.yaml:7]: I wonder if you could make this a matrix. I played around with something similar here.

https://github.com/beeme1mr/github-actions-experiments/blob/main/.github/workflows/js-test.yaml

Happy to collaborate with you if you're interested!

Comment (tcarrio) [reply .github/workflows/split_monorepo.yaml:7]: The issue I see is that [a matrix strategy is defined within a single job](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy), but each of these are defined as separate jobs.

I do want to eventually simplify this, but I really want to prove it satisfies the needs of this repository first. Theoretically `git-filter-repo` should work, but so did `gitsplit` at the time :slightly_smiling_face:

Comment (tcarrio) [reply .github/workflows/split_monorepo.yaml:7]: We should take a look at other approaches to this if all of this ends up actually working though. The entirety of each job would be be defined best as part of a single Action which would simplify a ton of this. `gitsplit` had a config file that would also be a good approach to this.

Comment (tcarrio) [reply .github/workflows/split_monorepo.yaml:7]: I am going to merge this just for the sake of testing that the `git-filter-repo` strategy works, and then we can work on making this more clean and DRY code.

**Comment by tcarrio** [comment]
Hold on to your butts

**Comment by tcarrio** [comment]
CURSES. It seems the changeset detection did not work for releases.

**Comment by tcarrio** [comment]
For reference, the change detection was for the release against the main branch... But it's cut from the main branch so the changeset will always be empty.

Boooo.

We might be able to do some fun stuff with Action State but definitely more than I had hoped for.

**Pull request merged**